app/controllers/application_controller.rb

```diff
@@ -0,0 +1,26 @@
+class ApplicationController < ActionController::Base
+  protect_from_forgery with: :exception
+  helper_method :categories, :current_order
+  before_action :remember_location
+
+  def categories
+    Category.all
+  end
+
+  def current_order
+    unless session[:order_id].nil?
```

Comment:
_unless ... else_ is very bad thing to do..
http://www.railstips.org/blog/archives/2008/12/01/unless-the-abused-ruby-conditional/

Never use `unless` with `else`. In you case, reverse the condition - do the simple if..else